File: app/src/main/java/com/wealoha/social/api/common/service/BaseService.java
package com.wealoha.social.api.common.service;

import java.util.List;

public interface BaseService<T> {

	public interface ServiceCallback {

		public void success();

		public void failer();
	}

	public interface ServiceResultCallback<T> {

		public void success(List<T> list);

		public void failer();

		public void nomore();

		public void beforeSuccess();

	}
	public interface ServiceObjResultCallback<T> {

		public void success(T obj);

		public void failer();

		public void nomore();

		public void beforeSuccess();

	}
	public interface ServiceListResultCallback<T> {

		public void success(List<T> t);

		public void failer();
	}

}

File: app/src/main/java/com/wealoha/social/api/BaseService.kt
package com.wealoha.social.api

interface BaseService<T> {
    interface ServiceCallback {
        fun success()
        fun failer()
    }

    interface ServiceResultCallback<T> {
        fun success(list: List<T>?)
        fun failer()
        fun nomore()
        fun beforeSuccess()
    }

    interface ServiceObjResultCallback<T> {
        fun success(obj: T)
        fun failer()
        fun nomore()
        fun beforeSuccess()
    }

    interface ServiceListResultCallback<T> {
        fun success(t: List<T>?)
        fun failer()
    }
}